src/audio/google/google_rtc_audio_processing.c

```diff
@@ -82,11 +95,49 @@ static int google_rtc_audio_processing_params(
 	struct sof_ipc_stream_params *params)
 {
 	int ret;
+#if CONFIG_IPC_MAJOR_4
+	struct google_rtc_audio_processing_comp_data *cd = comp_get_drvdata(dev);
+	struct comp_buffer __sparse_cache *sink_c;
+	struct comp_buffer *sink;
+
+	/* update sink buffer format */
+	memset(params, 0, sizeof(*params));
+	params->channels = cd->config.base_cfg.audio_fmt.channels_count;
+	params->rate = cd->config.base_cfg.audio_fmt.sampling_frequency;
+	params->sample_container_bytes = cd->config.base_cfg.audio_fmt.depth / 8;
+	params->sample_valid_bytes =
+		cd->config.base_cfg.audio_fmt.valid_bit_depth / 8;
+	params->buffer_fmt = cd->config.base_cfg.audio_fmt.interleaving_style;
+	params->buffer.size = cd->config.base_cfg.ibs;
+
+	/* update sink format */
+	if (!list_is_empty(&dev->bsink_list)) {
```

Comment:
shouldnt this be checked on create?

Reply:
when this module is created, no sink is attached. The sink buffer is attached when this module is connected to other modules.